R/methods-Chromatograms.R

```diff
@@ -190,3 +212,52 @@ setMethod("plot", signature = signature("Chromatograms"),
                            xlab = xlab, ylab = ylab, main = main, ...)
               }
           })
+
+#' @rdname Chromatograms-class
+#'
+#' @description \code{phenoData}: accesses the phenotypical desccription of the
+#'     samples. Returns an \code{NAnnotatedDataFrame} object.
+setMethod("phenoData", "Chromatograms", function(object) object@phenoData)
+
+#' @rdname Chromatograms-class
+#'
+#' @description \code{pData}: accesses the phenotypical description of the
+#'     samples. Returns a \code{data.frame}.
+setMethod("pData", "Chromatograms", function(object) pData(phenoData(object)))
+
+#' @rdname Chromatograms-class
+#'
+#' @description \code{pData<-}: replace the phenotype data.
+setReplaceMethod("pData", "Chromatograms", function(object, value) {
+    pData(object@phenoData) <- value
+    if (any(rownames(value) != as.character(1:nrow(value))))
+        colnames(object@.Data) <- rownames(value)
+    if (validObject(object))
+        object
+})
+
+#' @rdname Chromatograms-class
+#'
+#' @description \code{$} and \code{$<-}: get or replace individual columns of
+#'     the object's pheno data.
+#'
+#' @param name For \code{$}, the name of the pheno data column.
+setMethod("$", "Chromatograms", function(x, name) {
+    eval(substitute(pData(x)$NAME_ARG, list(NAME_ARG = name)))
+})
+#' @rdname Chromatograms-class
+setReplaceMethod("$", "Chromatograms", function(x, name, value) {
+    pData(x)[[name]] <- value
```

Comment:
In principle `pData(x)[[name]] <- ` throws an error if the lengths do not match, unless `value` is recycled. I did not check to mimic the *standard* R-behavior in such cases.